tests/cargo-add.rs

```diff
@@ -943,6 +943,95 @@ fn adds_dependency_with_target_cfg() {
     assert_eq!(val.as_str().unwrap(), "my-package1--CURRENT_VERSION_TEST");
 }
 
+#[test]
+fn adds_features_dependency() {
+    let (_tmpdir, manifest) = clone_out_test("tests/fixtures/add/Cargo.toml.sample");
+
+    // dependency not present beforehand
+    let toml = get_toml(&manifest);
+    assert!(toml["dependencies"].is_none());
+
+    execute_command(
+        &[
+            "add",
+            "https://github.com/killercup/cargo-edit.git",
+            "--features",
+            "jui",
+        ],
+        &manifest,
+    );
+
+    // dependency present afterwards
+    let toml = get_toml(&manifest);
+    let val = toml["dependencies"]["cargo-edit"]["features"][0].as_str();
+    assert_eq!(val, Some("jui"));
+}
+
+#[test]
+fn overrides_existing_features() {
+    overwrite_dependency_test(
+        &["add", "your-face", "--features", "nose"],
+        &["add", "your-face", "--features", "mouth"],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["mouth"] }
+"#,
+    )
+}
+
+#[test]
+fn can_be_forced_to_provide_an_empty_features_list() {
+    // FIXME: This should set `features = []`
+    overwrite_dependency_test(
+        &["add", "your-face"],
+        &["add", "your-face", "--features", ""],
+        r#"
+[dependencies]
+your-face = "your-face--CURRENT_VERSION_TEST"
+"#,
+    )
+}
+
+#[test]
+fn can_be_forced_to_remove_an_existing_features_list() {
+    // FIXME: I have no idea how we want to specify this on the command line.
```

Comment:
There is a decision to be made here. 

A)
* `cargo add your-face --features nose` sets features to ["nose"]
* `cargo add your-face` does not override any existing features (current behaviour)
* `cargo add your-face --features ""` sets features to []

This does not allow the user to remove `features = ...` from the command-line.

B)
* `cargo add your-face --features nose` sets features to ["nose"]
* `cargo add your-face` does not override any existing features (current behaviour)
* `cargo add your-face --features ""` wipes all features from your-face

This does not allow the user to set `features = []` from the command-line.

C) 
* `cargo add your-face --features nose` sets features to ["nose"]
* `cargo add your-face` wipes all features from your-face
* `cargo add your-face --features ""` sets features to []

This would break the "set a specific version of your-face, but keep features the same" workflow for existing users.

D) same as A) but with a --reset-features flag or similar, to remove `features = ...`?

I am going to aim for implementing A) in this PR, as it seems to be what people have agreed on before I arrived. If anyone has any bright ideas, leave them here and I will aim to ship them as a separate PR.

Reply:
**A** makes sense to me. If we have the following:

```toml
your-face = { version = "0.1", features = ["a", "b"] }
```

We can remove the feature `b` by doing something like the following, can't we?

```
cargo add your-face --features a
```

So, we don't need to do anything special regarding removing a feature.